services/src/main/java/org/fao/geonet/api/records/MetadataWorkflowApi.java

```diff
@@ -419,23 +442,268 @@ public MetadataStatus convertParameter(int id, MetadataStatusParameter parameter
+            checkCanViewStatus(fullMetadataText, metadataStatus, httpSession, request);
+        }
+
+        return fullMetadataText;
+    }
+
+    @ApiOperation(
+            value = "Restore saved content from a status record",
+            notes = "",
+            nickname = "restoreAtStatusSave")
+    @RequestMapping(
+            value = "/{metadataUuid}/status/{statusId:[0-9]+}.{userId:[0-9]+}.{changeDate}/restore",
+            method = RequestMethod.POST
+            )
+    @PreAuthorize("hasRole('Editor')")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Record restored."),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void restoreAtStatusSave(
+            @ApiParam(value = API_PARAM_RECORD_UUID, required = true) @PathVariable String metadataUuid,
+            @ApiParam(value = "Status identifier", required = true) @PathVariable int statusId,
+            @ApiParam(value = "User identifier", required = true) @PathVariable int userId,
+            @ApiParam(value = "Change date", required = true) @PathVariable String changeDate,
+            @ApiIgnore @ApiParam(hidden = true) HttpSession httpSession, HttpServletRequest request
+            )
+                    throws Exception {
+
+        ApplicationContext applicationContext = ApplicationContextHolder.get();
+        DataManager dataMan = applicationContext.getBean(DataManager.class);
+
+        MetadataStatus metadataStatus = metadataStatusRepository
+                .findOneByUuidAndStatusValue_IdAndUserIdAndChangeDate(metadataUuid, statusId, userId, new ISODate(changeDate));
+
+        if (metadataStatus == null) {
+            throw new ResourceNotFoundException(
+                    String.format("Can't find metadata status for record '%d', user '%s' at date '%s'. Record cannot be restored", metadataUuid,
+                            userId, changeDate));
+        }
+
+        if (!StatusValueType.event.equals(metadataStatus.getStatusValue().getType()) || !ArrayUtils.contains(supportedRestoreStatuses, statusId)) {
+            throw new NotAllowedException("The recover for this element is not supported. Supports status type '" +
+                    StatusValueType.event + "' with the status id '" + Arrays.toString(supportedRestoreStatuses) +
+                    "'. Received status type '" + metadataStatus.getStatusValue().getType() + "' with status id '" + statusId + "'");
+        }
+
+        if (metadataStatus.getPreviousState() == null) {
+            throw new ResourceNotFoundException(
+                    String.format("No data exists for previous state on metadata record '%d', user '%s' at date '%s'. Record cannot be restrored", metadataUuid,
+                            userId, changeDate));
+        }
+
+        AbstractMetadata metadata = null;
```

Comment:
To check to refactor with the code https://github.com/geonetwork/core-geonetwork/pull/4817/files#diff-668fdf0a09c1a8afb5771c9e85715598R496-R506, seem the same code

Reply:
refactored code for before/after api so that that they use common code.